Modules/Microsoft365DSC/DSCResources/MSFT_TeamsShiftsPolicy/MSFT_TeamsShiftsPolicy.psm1

```diff
@@ -0,0 +1,476 @@
+    $nullResult.Ensure = 'Absent'
+    try
+    {
+        $instance = Get-CsTeamsShiftsPolicy -Identity $Identity -ErrorAction SilentlyContinue
+        if ($null -eq $instance)
+        {
+            return $nullResult
+        }
+
+        Write-Verbose -Message "Found an instance with Identity {$Identity}"
+        $results = @{
+            Identity                       = $instance.Identity
+            AccessGracePeriodMinutes       = $instance.AccessGracePeriodMinutes
+            AccessType                     = $instance.AccessType
+            EnableScheduleOwnerPermissions = $instance.EnableScheduleOwnerPermissions
+            EnableShiftPresence            = $instance.EnableShiftPresence
+            ShiftNoticeFrequency           = $instance.ShiftNoticeFrequency
+            ShiftNoticeMessageCustom       = $instance.ShiftNoticeMessageCustom
+            ShiftNoticeMessageType         = $instance.ShiftNoticeMessageType
+            Ensure                         = 'Present'
+            Credential                     = $Credential
+            ApplicationId                  = $ApplicationId
+            TenantId                       = $TenantId
+            CertificateThumbprint          = $CertificateThumbprint
+        }
+        return [System.Collections.Hashtable] $results
+    }
+    catch
+    {
+        New-M365DSCLogEntry -Message 'Error retrieving data:' `
+            -Exception $_ `
+            -Source $($MyInvocation.MyCommand.Source) `
+            -TenantId $TenantId `
+            -Credential $Credential
+
+        return $nullResult
+    }
+}
+
+function Set-TargetResource
+{
+    [CmdletBinding()]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [System.String]
+        $Identity,
+
+        [Parameter()]
+        [System.Int64]
+        $AccessGracePeriodMinutes,
+
+        [Parameter()]
+        [System.String]
+        $AccessType,
+
+        [Parameter()]
+        [System.Boolean]
+        $EnableScheduleOwnerPermissions,
+
+        [Parameter()]
+        [System.Boolean]
+        $EnableShiftPresence,
+
+        [Parameter()]
+        [System.String]
+        $ShiftNoticeFrequency,
+
+        [Parameter()]
+        [System.String]
+        $ShiftNoticeMessageCustom,
+
+        [Parameter()]
+        [System.String]
+        $ShiftNoticeMessageType,
+
+        [Parameter()]
+        [ValidateSet('Present', 'Absent')]
+        [System.String]
+        $Ensure,
+
+        [Parameter()]
+        [System.Management.Automation.PSCredential]
+        $Credential,
+
+        [Parameter()]
+        [System.String]
+        $ApplicationId,
+
+        [Parameter()]
+        [System.String]
+        $TenantId,
+
+        [Parameter()]
+        [System.String]
+        $CertificateThumbprint
+    )
+
+    try
+    {
+        New-M365DSCConnection -Workload 'MicrosoftTeams' `
+            -InboundParameters $PSBoundParameters | Out-Null
+    }
+    catch
+    {
+        Write-Verbose -Message $_
```

Comment:
Same comment as before

Reply:
Yes, it should. This Teams resource was entirely generated via the Dynamic Resource Generator. I will regenerate with your comments included.